src/corehost/corehost.cpp

```diff
@@ -333,6 +337,51 @@ void buffering_trace_writer(const pal::char_t* message)
 {
     g_buffered_errors.append(message).append(_X("\n"));
 }
+
+// Determines if the current module (should be the apphost.exe) is marked as Windows GUI application
+// in case it's not a GUI application (so should be CUI) or in case of any error the function returns false.
+bool get_windows_graphical_user_interface_bit()
+{
+    // The first two bytes of a PE file are a constant signature.
+    const UINT16 PEFileSignature = 0x5A4D;
+
+    // The offset of the PE header pointer in the DOS header.
+    const int PEHeaderPointerOffset = 0x3C;
+
+    // The offset of the Subsystem field in the PE header.
+    const int SubsystemOffset = 0x5C;
```

Comment:
This offset is different between 32-bit and 64-bit systems. `IMAGE_OPTIONAL_HEADER`

Reply:
Actually the offset is the same (I tried with offsetof and so on). I will change the code here to use the structs from windows headers. This offset (actually the entire method) was copied from the SDK's code which marks the apphost as GUI app, so I will leave that code as is (it's C# and thus has no access to the structs, but the offsets should be correct).